--- a/src/vulkan/wsi/wsi_common_drm.c   2025-01-22 12:12:23.000000000 -0600
+++ b/src/vulkan/wsi/wsi_common_drm.c   2025-02-21 21:52:48.716295323 -0600
@@ -371,8 +371,12 @@ wsi_create_sync_for_image_syncobj(const
 
       if (drmSyncobjTransfer(device->drm_fd, tmp_handles[i], 0,
                              image->explicit_sync[i].handle, image->explicit_sync[i].timeline, 0)) {
-         result = vk_errorf(NULL, VK_ERROR_OUT_OF_DEVICE_MEMORY, "Failed to transfer syncobj. Was the timeline point materialized? Errno: %d - %s", errno, strerror(errno));
-         goto fail;
+         for (uint32_t i = 0; i < WSI_ES_COUNT; i++) {
+            if (tmp_handles[i])
+               drmSyncobjDestroy(device->drm_fd, tmp_handles[i]);
+         }
+
+         return wsi_create_sync_imm(device, sync_out);
       }
       if (drmSyncobjExportSyncFile(device->drm_fd, tmp_handles[i], &sync_file_fds[i])) {
          result = vk_errorf(NULL, VK_ERROR_OUT_OF_DEVICE_MEMORY, "Failed to export sync file. Errno: %d - %s", errno, strerror(errno));