diff --git a/dlls/ntdll/unix/signal_x86_64.c b/dlls/ntdll/unix/signal_x86_64.c
index e0eddb8..d5dd22b 100644
--- a/dlls/ntdll/unix/signal_x86_64.c
+++ b/dlls/ntdll/unix/signal_x86_64.c
@@ -535,7 +535,7 @@ static inline struct amd64_thread_data *amd64_thread_data( struct thread_data *d
 static unsigned int frame_size;
 static unsigned int xstate_size = sizeof(XSAVE_AREA_HEADER);
 static UINT64 xstate_extended_features;
-static LONG syscall_dispatch_enabled = TRUE;
+static LONG syscall_dispatch_enabled = FALSE;
 
 static inline struct thread_data *get_current_thread_data(void)
 {
