From eb6c207ce5910123ded4aacb0f52d6804cb68251 Mon Sep 17 00:00:00 2001
From: mactan-sc <77816662+mactan-sc@users.noreply.github.com>
Date: Tue, 27 Jan 2026 17:26:08 -0600
Subject: [PATCH] Dummy dlls to satisfy some checks

---
 loader/wine.inf.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 5bdb6455a06..557096045af 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -831,0 +832,3 @@ HKLM,SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\x86,"Version",2,"14.42.344
+; dummy dlls
+11,,devobj.dll,security.dll
+11,,drvstore.dll,security.dll
-- 
2.52.0

